litestar/response/base.py

```diff
@@ -24,28 +22,172 @@
         ResponseHeaders,
         Scope,
         Send,
+        Serializer,
         TypeEncodersMap,
     )
 
+__all__ = ("ASGIResponse", "Response")
+
 T = TypeVar("T")
 
 
+class ASGIResponse:
+    """A low-level ASGI response class."""
+
+    __slots__ = (
+        "background",
+        "body",
+        "content_length",
+        "encoded_headers",
+        "encoding",
+        "is_head_response",
+        "status_code",
+    )
+
+    def __init__(
+        self,
+        *,
+        background: BackgroundTask | BackgroundTasks | None = None,
+        body: bytes = b"",
```

Comment:
I think this can be `str` as well. Makes the usage a bit more convenient and aligns with all the low-level de/encoders we have.